hilti/toolchain/src/compiler/visitors/coercer.cc

```diff
@@ -130,6 +132,41 @@ struct Visitor : public visitor::PreOrder<void, Visitor> {
             return {std::nullopt};
     }
 
+    /**
+     * Coerces a specific call argument to a given typel returning the coerced
+     * expression (only) if its type has changed.
+     */
+    Result<std::optional<Expression>> coerceMethodArgument(const expression::ResolvedOperatorBase& o, size_t i,
+                                                           const Type& t) {
+        auto ops = o.op2();
+
+        // If the argument list was the result of a coercion unpack its result.
+        if ( auto coerced = ops.tryAs<expression::Coerced>() )
+            ops = coerced->expression();
+
+        auto ctor_ = ops.as<expression::Ctor>().ctor();
+
+        // If the argument was the result of a coercion unpack its result.
+        if ( auto x = ctor_.tryAs<ctor::Coerced>() )
+            ctor_ = x->coercedCtor();
+
+        const auto& args = ctor_.as<ctor::Tuple>().value();
+        if ( i >= args.size() )
+            return {std::nullopt};
+
+        auto narg = hilti::coerceExpression(args[i], t);
+        if ( ! narg )
+            return result::Error(fmt("cannot coerce argument %d from %s to %s", i, args[i].type(), t));
+
+        if ( narg.nexpr ) {
+            auto nargs = args.copy();
+            nargs[i] = *narg.nexpr;
+            return {expression::Ctor(ctor::Tuple(nargs))};
+        }
```

Comment:
We could make scoping slightly clearer here, e.g.,
```suggestion
        if ( auto narg = hilti::coerceExpression(args[i], t); ! narg )
            return result::Error(fmt("cannot coerce argument %d from %s to %s", i, args[i].type(), t));

        else if ( narg.nexpr ) {
            auto nargs = args.copy();
            nargs[i] = *narg.nexpr;
            return {expression::Ctor(ctor::Tuple(nargs))};
        }

```